app/src/main/java/com/cornellappdev/android/eateryblue/ui/viewmodels/EateryDetailViewModel.kt

```diff
@@ -24,32 +30,28 @@ class EateryDetailViewModel @Inject constructor(
 ) : ViewModel() {
     private val eateryId: Int = checkNotNull(savedStateHandle["eateryId"])
 
-    var eateryRetrievalState: EateryRetrievalState by mutableStateOf(EateryRetrievalState.Pending)
-        private set
+    var eatery: StateFlow<EateryApiResponse<Eatery>> = eateryRepository.eateryFlow.map { eateries ->
+        when (eateries) {
+            is EateryApiResponse.Pending -> EateryApiResponse.Pending
+
+            is EateryApiResponse.Error -> EateryApiResponse.Error
 
-    var eatery by mutableStateOf(Eatery())
+            is EateryApiResponse.Success -> {
+                EateryApiResponse.Success(eateries.data.first {
+                    it.id == eateryId
+                })
+            }
+        }
+    }.stateIn(
+        CoroutineScope(Dispatchers.Main),
```

Comment:
Why are you using the main UI thread here, I think you should use `viewModelScope` no?

Reply:
Ouu, did not know that was a thing... that's DEFINITELY correct, yes, and I'll have to go change that in like 6 different places in Uplift too LOL. Thanks for the catch. Fixed in [e791ff5](https://github.com/cuappdev/eatery-blue-android/pull/48/commits/e791ff54fe9696bd61b327bd60e02c56db56babe).